### PR TITLE
[#703] update pipelines for agents support

### LINKED ISSUE #703: Allow jenkins jobs to be executed at AWS agent nodes
As a devops engineer I'd like to be able to run jenkins jobs at slave nodes so I can distribute the load across several instances

### PR DESCRIPTION
**Pull request by alexander-semenets**
this fixes #703 

This PR makes all ci stages to be executable at dynamically provisioned EC2 agents in isolated docker containers. There is also a portion of ansible playbooks for ec2 agent's ami setup pushed to the legion ci repository.

REMINDER:
- CI Jobs should be enabled at the new Jenkins master
- cc. dns record should be pointed to the new address
- ci-infra tests tags should be restored to "infra"

### CODE REVIEW

File: legion_test/requirements/Pipfile [hunk 1]
```diff
@@ -20,11 +20,13 @@ Pillow = "==5.0.0"
...
 PyYAML = "==3.12"
 PyJWT = "==1.6.4"
-"boto3" = "==1.7.17"
+"boto3" = "==1.9.75"
+"botocore" = "==1.12.75"
 python-jenkins = "==0.4.15"
 robotframework = "==3.0.2"
 robotframework-seleniumlibrary = "==3.1.1"
 robotframework-pabot = "==0.44"
+slackclient = "==1.2.1"
```
Comment (kirillmakhonin) [inline legion_test/requirements/Pipfile]: This line is invalid - `slackclient` is not required for this package

File: pipeline.Dockerfile [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
 # Install additional tools for build purposes
-RUN pip install Sphinx==1.8.0 sphinx_rtd_theme==0.4.1 sphinx-autobuild==0.7.1 recommonmark==0.4.0 twine==1.11.0 ansible==2.6.4 awscli==1.16.19
+RUN pip install Sphinx==1.8.0 sphinx_rtd_theme==0.4.1 sphinx-autobuild==0.7.1 \
+  recommonmark==0.4.0 twine==1.11.0 awscli==1.16.85 ansible==2.7.2 yq==2.7.1
 
 # Add sources
-
-
 ADD legion /src/legion
+RUN cd /src/legion/requirements && pipenv install --system --dev
```
Comment (kirillmakhonin) [inline pipeline.Dockerfile]: Please move requirements installation back to previous layers (for caching purposes)

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
ok, to test

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test all

File: pipeline.Dockerfile (hunk 2 above)
Comment (alexander-semenets) [reply pipeline.Dockerfile]: done

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
image

**Pull request merged**